### ISSUE TITLE
BaseModel.find callback issue

### ISSUE THREAD

**Issue by edimoldovan** [jedireza/hapi-mongo-models#4]
Hi,

I have found an interesting issue, I thought I would mention it here. 

On line https://github.com/jedireza/hapi-mongo-models/blob/master/lib/base-model.js#L244 my find methond worked onny if I replaced the above line with this: 

var callback = args.pop();

I am not sure why the original line doesn’t work, but the symptom is that it returns a good result with one issue: elements in the data array are all empty objects. 

Happy to provide more detail if needed.

**Comment by jedireza** [comment]
Yeah we need that callback to be send through the `resultFactory`. Would you be willing to create a gist that reproduces the error so I can help debug?

**Comment by edimoldovan** [comment]
Yes, of course, I’ll try to put that together today or tomorrow. Thx for the reply!

**Comment by edimoldovan** [comment]
I prepared a repo instead, with two example calls which reproduce the issue: https://github.com/edimoldovan/hapi-mongo-models-example

If you do this: 
```
curl -X POST -H "Content-Type: application/json; charset=utf-8" "127.0.0.1:8000/api/users" -d '{"email":"pistike@pistike.hu","password":"bla","name":"name”}’
```
a user will be inserted, but not returned in the response. Also, if you hit this url: 
```
http://localhost:8000/api/users
```
you will see that the data array contains the correct number of objects, only they are empty.

If I do the change I mentioned above, all this suddenly becomes ok.

**Comment by jedireza** [comment]
Thanks so much. That made it much easier to help figure out why the objects were empty for you.

Your model was missing a `constructor`. When we create a new instance of a model we pass the instance data into it's `constructor`.

Here is what I did to fix your example.

```bash
$ npm install object-assign --save
```

Then in your model I added a constructor:

```js
var Joi = require('joi');
var BaseModel = require('hapi-mongo-models').BaseModel;
var ObjectAssign = require('object-assign');

var User = BaseModel.extend({
    constructor: function (attrs) {

        ObjectAssign(this, attrs);
    }
});

User._collection = 'users'; // the mongo collection name

User.schema = Joi.object().keys({
	name: Joi.string(),
	image: Joi.string(),
	password: Joi.string().required(),
	email: Joi.string().email().required()
});

module.exports = User;
```

Then both `GET` and `POST` worked as expected.
